sharding/validator.go

```diff
@@ -51,6 +51,16 @@ func (v *validator) SetChances(chances uint32) {
 	v.mutChances.Unlock()
 }
 
+// Clone clones the validator
+func (v *validator) Clone() (interface{}, error) {
+	if v == nil {
+		return nil, ErrNilValidator
+	}
+	v2 := *v
```

Comment:
This will not copy the Pk and Address which are []byte but the pointer to them. Should it copy them? Could the PK and Address get changed like v.PK()[7] = 5? If not there is no need to copy it.

Reply:
ok

Reply:
Correct it will not deep copy the Pk and Addr but this is fine as they will not change.